OpenTrons/opentrons-api
# refactor(api): Drop tip after finish behavior removal

# Overview
 With the addition of the tip drop wizard, we no longer should be automatically dropping tips at the end of protocols. Instead, users should explicitly state in their protocols to drop tips. Alternatively, if there is still a tip attached at the end of a protocol finish then the user should be prompted on the ODD and/or app to manually drop the tip utilizing the drop tip wizard, and be given the choice to keep the tip.

# Test Plan
The test plan will be the same for Flex and OT2 at this current state. The test plan is as follows:
1. Load a protocol on a Robot that picks up a tip without a drop tip at the end.
2. Ensure the Robot keeps the tip after the protocol finishes.
3. Load a protocol on a Robot with a drop tip command at the end.
4. Ensure the Robot discards the tip at the end of the protocol before finishing.

# Risk assessment
There is a risk that users utilizing older protocols on OT2 will expect the tip to automatically drop at the end of their protocol. We can put a statement on the latest API version that states that protocols will no longer automatically drop tips upon completion, or possibly prompt users with a warning during analysis.
